src/test/java/org/cactoos/collection/CollectionOfTest.java

```diff
@@ -79,4 +87,132 @@ public void testToStringEmpty() {
+        new Assertion<>(
+            "Collection is equal to a collection with different content",
+            new CollectionOf<>("a", "b"),
+            new IsNot<>(new IsEqual<>(new CollectionOf<>("a", "c")))
+        ).affirm();
+    }
+
+    @Test
+    public void equalToItself() {
+        final CollectionOf<String> col = new CollectionOf<>("val1", "val2");
+        new Assertion<>(
+            "Collection is not equal to itself",
+            col,
+            new IsEqual<>(col)
+        ).affirm();
+    }
+
+    @Test
+    public void equalToCollectionWithIdenticalContent() {
+        new Assertion<>(
+            "Collection is not equal to a collection with identical content",
+            new CollectionOf<>("val1", "val2"),
+            new IsEqual<>(new CollectionOf<>("val1", "val2"))
+        ).affirm();
+    }
+
+    @Test
+    public void equalToListWithIdenticalContent() {
+        new Assertion<>(
+            "Collection not equal to a list with identical content",
+            new CollectionOf<>("a"),
+            new IsEqual<>(new ListOf<>("a"))
+        ).affirm();
+    }
+
+    @Test
+    public void equalToEmptyCollection() {
+        new Assertion<>(
+            "Empty collection not equal with empty collection",
+            new CollectionOf<>(),
+            new IsEqual<>(new CollectionOf<>())
+        ).affirm();
+    }
+
+    @Test
+    public void notEqualToNull() {
+        new Assertion<>(
+            "Empty collection equal to null",
+            new CollectionOf<>(),
+            new IsNot<>(new IsEqual<>(null))
+        ).affirm();
+    }
+
+    @Test
+    public void hashCodeEqual() {
+        new Assertion<>(
+            "HashCode returns different results for same entries",
+            new CollectionOf<>("a", "b").hashCode(),
+            new IsEqual<>(new CollectionOf<>("a", "b").hashCode())
+        ).affirm();
+    }
+
+    @Test
+    public void differentHashCode() {
+        new Assertion<>(
+            "HashCode returns identical results for different entries",
+            new CollectionOf<>("a", "b").hashCode(),
+            new IsNot<>(new IsEqual<>(new CollectionOf<>("b", "a").hashCode()))
+        ).affirm();
+    }
+
+    @Test
+    public void emptyCollectionEnvelopeShouldBeEqualToEmptyCollection() {
+        final Collection<String> envelope = new CollectionOf<>();
+        final Collection<String> collection = Collections.emptyList();
+        new Assertion<>(
+            "Empty envelope and collection should be equal",
+            envelope,
+            new IsEqual<>(collection)
+        ).affirm();
+    }
+
+    @Test
+    public void collectionEnvelopeShouldBeEqualToCollection() {
+        final Collection<String> envelope = new CollectionOf<>("a");
+        final Collection<String> collection = new ArrayList<>(1);
+        collection.add("a");
+        new Assertion<>(
+            "Envelope and collection should be equal",
```

Comment:
@victornoel same above (with properly adaptation).

Reply:
@fabriciofx actually I removed this one because the test is redundant with one of the above and it has nothing to do with envelope anymore

Reply:
@victornoel ok